x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyExecutor.java

```diff
@@ -51,8 +90,17 @@ public void runPolicy(String policyId, ActionListener<PolicyExecutionResult> lis
     }
 
     public void runPolicy(String policyName, EnrichPolicy policy, ActionListener<PolicyExecutionResult> listener) {
-        EnrichPolicyRunner runnable = new EnrichPolicyRunner(policyName, policy, listener, clusterService, client,
-            indexNameExpressionResolver, nowSupplier, fetchSize);
-        threadPool.executor(ThreadPool.Names.GENERIC).execute(runnable);
+        if (tryLockingPolicy(policyName)) {
```

Comment:
this might be tangential to this review, but should we also cap to total number of concurrent policies running ? 

By default the GENERIC threadpool will scale up to 512 !

Reply:
I think this is definitely a separate PR. We'll probably want that to be a soft limit backed by a config.